src/main/java/mSearch/filmlisten/FilmlisteLesen.java

```diff
@@ -119,125 +82,187 @@ private InputStream selectDecompressor(String source, InputStream in) throws Exc
+                }
+                if (datenFilm.arr[DatenFilm.FILM_THEMA].isEmpty()) {
+                    datenFilm.arr[DatenFilm.FILM_THEMA] = thema;
+                } else {
+                    thema = datenFilm.arr[DatenFilm.FILM_THEMA];
+                }
 
-                    listeFilme.importFilmliste(datenFilm);
-                    if (seconds > 0) {
-                        // muss "rückwärts" laufen, da das Datum sonst 2x gebaut werden muss
-                        // wenns drin bleibt, kann mans noch ändern
-                        if (!checkDate(datenFilm)) {
-                            listeFilme.remove(datenFilm);
-                        }
+                listeFilme.importFilmliste(datenFilm);
+                if (milliseconds > 0) {
+                    // muss "rückwärts" laufen, da das Datum sonst 2x gebaut werden muss
+                    // wenns drin bleibt, kann mans noch ändern
+                    if (!checkDate(datenFilm)) {
+                        listeFilme.remove(datenFilm);
                     }
                 }
             }
+        }
+    }
+
+    /**
+     * Read a locally available filmlist.
+     *
+     * @param source     file path as string
+     * @param listeFilme the list to read to
+     */
+    private void processFromFile(String source, ListeFilme listeFilme) {
+        notifyProgress(source, PROGRESS_MAX);
+        try (InputStream in = selectDecompressor(source, new FileInputStream(source));
+             JsonParser jp = new JsonFactory().createParser(in)) {
+            readData(jp, listeFilme);
         } catch (FileNotFoundException ex) {
             Log.errorLog(894512369, "FilmListe existiert nicht: " + source);
             listeFilme.clear();
         } catch (Exception ex) {
             Log.errorLog(945123641, ex, "FilmListe: " + source);
             listeFilme.clear();
         }
+    }
+
+    private void checkDays(long days) {
+        if (days > 0) {
+            milliseconds = System.currentTimeMillis() - TimeUnit.MILLISECONDS.convert(days, TimeUnit.DAYS);
+        } else {
+            milliseconds = 0;
+        }
+    }
 
-        if (Config.getStop()) {
-            Log.sysLog("--> Abbruch");
+    public void readFilmListe(String source, final ListeFilme listeFilme, int days) {
+        try {
+            Log.sysLog("Liste Filme lesen von: " + source);
             listeFilme.clear();
+            this.notifyStart(source, PROGRESS_MAX); // für die Progressanzeige
+
+            checkDays(days);
+
+            if (!source.startsWith("http")) {
+                processFromFile(source, listeFilme);
+            } else {
+                processFromWeb(new URL(source), listeFilme);
+            }
+
+            if (Config.getStop()) {
+                Log.sysLog("--> Abbruch");
+                listeFilme.clear();
+            }
+        } catch (MalformedURLException ex) {
+            ex.printStackTrace();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)